typing/includemod.ml

```diff
@@ -288,6 +288,18 @@ and signatures env cxt subst sig1 sig2 =
             ((id,pos,Tcoerce_none)::l , pos+1)
         | item -> (l, if is_runtime_component item then pos+1 else pos))
       ([], 0) sig1 in
+
+  let runtime_fields =
+    let get_id = function
+      | Sig_value (i,_)
+      | Sig_module (i,_,_)
+      | Sig_typext (i,_,_)
+      | Sig_modtype(i,_)
+      | Sig_class (i,_,_)
+      | Sig_class_type(i,_,_)
+      | Sig_type(i,_,_) -> Ident.name i in
+      sig2 |> (List.filter is_runtime_component) |> (List.map get_id) in
```

Comment:
can we use fold_left/right to avoid two passes

Reply:
done